src/binding/petsc4py/src/petsc4py/PETSc/DMDA.pyx

```diff
@@ -463,51 +1112,64 @@ cdef class DMDA(DM):
     #
 
     property dim:
-        def __get__(self):
+        """The dimension."""
+        def __get__(self) -> int:
             return self.getDim()
 
     property dof:
-        def __get__(self):
+        """The number of dof associated with each stratum of the grid."""
+        def __get__(self) -> int:
             return self.getDof()
 
     property sizes:
-        def __get__(self):
+        """The global dimension in each direction."""
+        def __get__(self) -> tuple[int, ...]:
             return self.getSizes()
 
     property proc_sizes:
-        def __get__(self):
+        """The number of ranks in each direction in the global decomposition."""
```

Comment:
I don't like `ranks`, I prefer `processes`. Maybe something we should make uniform throughout the docs. @dalcinl ?

Reply:
Go ahead.